internal/cfgmgr/cfgmgr.go

```diff
@@ -0,0 +1,149 @@
+package cfgmgr
+
+import (
+	"fmt"
+	"strings"
+	"unicode/utf16"
+
+	"github.com/Microsoft/go-winio/pkg/guid"
+	"github.com/pkg/errors"
+)
+
+//go:generate go run ../../mksyscall_windows.go -output zsyscall_windows.go cfgmgr.go
+
+//sys cmGetDeviceIDListSize(pulLen *uint32, pszFilter *byte, uFlags uint64) (hr error) = cfgmgr32.CM_Get_Device_ID_List_SizeA
+//sys cmGetDeviceIDList(pszFilter *byte, buffer *byte, bufferLen uint64, uFlags uint64) (hr error)= cfgmgr32.CM_Get_Device_ID_ListA
+//sys cmLocateDevNode(pdnDevInst *uint32, pDeviceID string, uFlags uint64) (hr error) = cfgmgr32.CM_Locate_DevNodeW
+//sys cmGetDevNodeProperty(dnDevInst uint32, propertyKey *devPropKey, propertyType *uint64, propertyBuffer *uint16, propertyBufferSize *uint64, uFlags uint64) (hr error) = cfgmgr32.CM_Get_DevNode_PropertyW
+
+type devPropKey struct {
+	fmtid guid.GUID
+	pid   uint64
+}
+
+const (
+	CM_GETIDLIST_FILTER_NONE               = uint64(0x00000000)
+	CM_GETIDLIST_FILTER_ENUMERATOR         = uint64(0x00000001)
+	CM_GETIDLIST_FILTER_SERVICE            = uint64(0x00000002)
+	CM_GETIDLIST_FILTER_EJECTRELATIONS     = uint64(0x00000004)
+	CM_GETIDLIST_FILTER_REMOVALRELATIONS   = uint64(0x00000008)
+	CM_GETIDLIST_FILTER_POWERRELATIONS     = uint64(0x00000010)
+	CM_GETIDLIST_FILTER_BUSRELATIONS       = uint64(0x00000020)
+	CM_GETIDLIST_DONOTGENERATE             = uint64(0x10000040)
+	CM_GETIDLIST_FILTER_TRANSPORTRELATIONS = uint64(0x00000080)
+	CM_GETIDLIST_FILTER_PRESENT            = uint64(0x00000100)
+	CM_GETIDLIST_FILTER_CLASS              = uint64(0x00000200)
+	CM_GETIDLIST_FILTER_BITS               = uint64(0x100003FF)
+
+	CM_LOCATE_DEVNODE_NORMAL       = uint64(0x00000000)
+	CM_LOCATE_DEVNODE_PHANTOM      = uint64(0x00000001)
+	CM_LOCATE_DEVNODE_CANCELREMOVE = uint64(0x00000002)
+	CM_LOCATE_DEVNODE_NOVALIDATION = uint64(0x00000004)
+
+	DEVPROP_TYPE_STRING        = uint64(0x00000012)
+	DEVPROP_TYPEMOD_LIST       = uint64(0x00002000)
+	DEVPROP_TYPE_STRING_LIST   = uint64(DEVPROP_TYPE_STRING | DEVPROP_TYPEMOD_LIST)
+	DEVPKEY_LOCATIONPATHS_GUID = "a45c254e-df1c-4efd-8020-67d146a850e0"
+
+	STRING_BUFFER_SIZE = uint64(1024)
+)
+
+func getDevPKeyDeviceLocationPaths() (*devPropKey, error) {
+	guid, err := guid.FromString(DEVPKEY_LOCATIONPATHS_GUID)
+	if err != nil {
+		return nil, err
+	}
+	return &devPropKey{
+		fmtid: guid,
+		pid:   37,
```

Comment:
A comment explaining what `37` means here, and where it is defined, would be good.